version: "{build}"

branches:

  only:

  - master

clone_depth: 1

build: false

environment:
    DEBUG: True
    REDIRECT_URIS: http://test.example.com:8000/api/pulse/oauth2callback
    SSL_PROTECTION: False
    DATABASE_URL: postgres://postgres:Password12!@localhost:5432/pulseapi
    SECRET_KEY: mydummykey
    AUTH_STAFF_EMAIL_DOMAINS: mozillafoundation.org

    matrix:
      - PYTHON: "C:\\Python37"
        PYTHON_VERSION: "3.7.x"
        PYTHON_ARCH: "64"

init:
  - "ECHO %PYTHON% %PYTHON_VERSION% %PYTHON_ARCH%"

install:
  - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
  - "python -m pip install --upgrade pip"
  - "python -m venv venv"
  - ".\\venv\\Scripts\\activate"
  - "pip install -r requirements.txt -r dev-requirements.txt"

services:
  - postgresql

before_test:
  - SET PGUSER=postgres
  - SET PGPASSWORD=Password12!
  - PATH=C:\Program Files\PostgreSQL\9.6\bin\;%PATH%
  - createdb pulseapi
  - "python manage.py migrate"

test_script:
  - "flake8 pulseapi"
  - "python manage.py test"

cache:
  - "%LOCALAPPDATA%\\pip\\Cache"
